include/osdp.h

```diff
@@ -943,6 +932,29 @@ int osdp_cp_modify_flag(osdp_t *ctx, int pd_idx, uint32_t flags, bool do_set);
 /*          Common Methods         */
 /* ------------------------------- */
 
+/**
+ * @brief Callback for command completion event callbacks. After it has
+ * been registered with `osdp_set_command_complete_callback()` this method is
+ * invoked after a command has been processed successfully in CP and PD sides.
+ *
+ * @param arg pointer that was passed to the arg param of
+ * `osdp_set_command_complete_callback`.
+ * @param id OSDP command ID (Note: this is not `enum osdp_cmd_e`)
+ */
+typedef void (*osdp_command_complete_callback_t)(void *arg, int id);
+
+/**
+ * @brief Set callback method for command completion notification.
+ * This callback is invoked when a reply to a command is received.
+ *
+ * @param ctx OSDP context
+ * @param cb The callback function's pointer
+ * @param arg A pointer that will be passed as the first argument of `cb`
+ */
+void osdp_set_command_complete_callback(osdp_t *ctx,
+					osdp_command_complete_callback_t cb,
+					void *arg);
+
```

Comment:
This API was marked deprecated and then removed [here](https://github.com/goToMain/libosdp/commit/0bde480c89fde2b317f2fad712333a3211b9b7eb#diff-8c3f6bb64d22c45f45f9f9735b80c1a6a104e7e9b369dd4a121cf8a03e0ffa22L1055-L1070) in preparation for a major release (v3.0.0) from master. But I see the removal was done hastily and left most of the implementation :(. I will remove the reset of this API in a bit.

Curious, what is your use of this API? It exposes low level OSDP commands to the user (which they should not have to care about). If you have a valid use-case, we can always expose a different API to cover that case.

Reply:
Just came across this yesterday, the use case in our application  currently need to kickoff some application specific tasks when CP receives a response for some OSDP command e.g. REPLY_PDID, REPLY_PDCAP..

Reply:
No direct use case at the moment.  I just noticed the inconsistency when I was tracing through the callback flows.